cestzam-core/src/main/java/be/valuya/cestzam/core/auth/itsme/ItsmeSessionHandler.java

```diff
@@ -0,0 +1,47 @@
+package be.valuya.cestzam.core.auth.itsme;
+
+import be.valuya.cestzam.core.auth.CestzamAuthHandler;
+import be.valuya.cestzam.core.session.CestzamSession;
+import be.valuya.cestzam.core.session.cookie.CestzamCookies;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class ItsmeSessionHandler implements CestzamAuthHandler {
+
+    @Override
+    public String getMethod() {
+        return ItsmeAuthSession.ITSME;
+    }
+
+    @Override
+    public CestzamSession processAuthSession(CestzamSession cestzamSession) {
+        ItsmeAuthSession authSession = (ItsmeAuthSession) cestzamSession.getAuthSession();
+        ItsmeAuthSessionStatus status = authSession.getStatus();
+        return switch (status) {
+            case NEW_SESSSION -> handleNewSession(cestzamSession, (ItsmeAuthSession.New) authSession);
+            case POKA_YOKA_CHALLENGE_REQUEST ->
+                    handleCodeRequestSesssion(cestzamSession, (ItsmeAuthSession.PokaYokaChallengeRequest) authSession);
+            case AUTHENTICATED -> cestzamSession;
+            case EXPIRED -> cestzamSession;
+            case ERRORED -> cestzamSession;
+        };
+    }
+
+    private CestzamSession handleNewSession(CestzamSession cestzamSession, ItsmeAuthSession.New authSession) {
+        CestzamCookies cookies = cestzamSession.getCookies();
+        String phoneNumber = authSession.getPhoneNumber();
+
+        // TODO submit username/password form & wait for mail at configuredRecipientEmail
```

Comment:
The user clicks on 'itsme' and submits the phone number.

if poka yoka is required, the image url is returned